File: android/app/src/main/kotlin/com/stillmoment/domain/models/ResolvedSoundscape.kt
package com.stillmoment.domain.models

/**
 * A resolved soundscape audio entry, regardless of source (built-in or user-imported).
 *
 * Consumers use this instead of checking BackgroundSoundRepository + CustomAudioRepository separately.
 * The resolver transparently handles both sources.
 *
 * @property id Unique identifier (BackgroundSound.id for built-in, UUID string for custom)
 * @property displayName Localized display name
 */
data class ResolvedSoundscape(
    val id: String,
    val displayName: String
)

File: ios/StillMoment/Domain/Models/ResolvedSoundscape.swift
//
//  ResolvedSoundscape.swift
//  Still Moment
//
//  Domain - Resolved soundscape audio (built-in or custom)
//

import Foundation

/// A resolved soundscape audio entry, regardless of source (built-in or user-imported).
///
/// Consumers use this instead of checking BackgroundSoundRepository + CustomAudioRepository separately.
/// The resolver transparently handles both sources.
struct ResolvedSoundscape: Equatable {
    /// Unique identifier (BackgroundSound.id for built-in, UUID string for custom)
    let id: String

    /// Localized display name
    let displayName: String
}
